RecoTracker/MkFitCMS/interface/LayerNumberConverter.h

```diff
@@ -0,0 +1,172 @@
+        //TIB
+        if (cmsswdet == 3 && cmsswlay == 1 && isStereo == -1)
+          return 3 + lOffset;
+        if (cmsswdet == 3 && cmsswlay == 2 && isStereo == -1)
+          return 4 + lOffset;
+        if (cmsswdet == 3 && cmsswlay == 3 && isStereo == 0)
+          return 5 + lOffset;
+        if (cmsswdet == 3 && cmsswlay == 4 && isStereo == 0)
+          return 6 + lOffset;
+        //TOB
+        if (cmsswdet == 5 && cmsswlay == 1 && isStereo == -1)
+          return 7 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 2 && isStereo == -1)
+          return 8 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 3 && isStereo == 0)
+          return 9 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 4 && isStereo == 0)
+          return 10 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 5 && isStereo == 0)
+          return 11 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 6 && isStereo == 0)
+          return 12 + lOffset;
+        return -1;
+      } else {
+        //TIB
+        if (cmsswdet == 3 && cmsswlay == 1 && isStereo == 0)
+          return 3 + lOffset;
+        if (cmsswdet == 3 && cmsswlay == 1 && isStereo == 1)
+          return 4 + lOffset;
+        if (cmsswdet == 3 && cmsswlay == 2 && isStereo == 0)
+          return 5 + lOffset;
+        if (cmsswdet == 3 && cmsswlay == 2 && isStereo == 1)
+          return 6 + lOffset;
+        if (cmsswdet == 3 && cmsswlay == 3 && isStereo == 0)
+          return 7 + lOffset;
+        if (cmsswdet == 3 && cmsswlay == 4 && isStereo == 0)
+          return 8 + lOffset;
+        //TOB
+        if (cmsswdet == 5 && cmsswlay == 1 && isStereo == 0)
+          return 9 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 1 && isStereo == 1)
+          return 10 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 2 && isStereo == 0)
+          return 11 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 2 && isStereo == 1)
+          return 12 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 3 && isStereo == 0)
+          return 13 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 4 && isStereo == 0)
+          return 14 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 5 && isStereo == 0)
+          return 15 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 6 && isStereo == 0)
+          return 16 + lOffset;
+        return -1;
+      }
+    }
+    int convertDiskNumber(int cmsswdet, int cmsswdisk, bool useMatched, int isStereo) const {
+      if (cmsswdet == 1 || cmsswdet == 3 || cmsswdet == 5)
+        return -1;  //BPIX, TIB, TOB
+      if (cmsswdet == 2)
+        return cmsswdisk - 1;  //FPIX
+      int lOffset = 0;
+      if (lo_ == TkLayout::phase1)
+        lOffset = 1;
+      if (useMatched) {
+        return -1;
+      } else {
+        //TID
+        if (cmsswdet == 4 && cmsswdisk == 1 && isStereo == 0)
+          return 2 + lOffset;
+        if (cmsswdet == 4 && cmsswdisk == 1 && isStereo == 1)
+          return 3 + lOffset;
```

Comment:
wouldn't it make sense to optimize this branching with `else if` to exclude repeated checking of the same values?

```suggestion
        if (cmsswdet == 4 && cmsswdisk == 1) {
          if (isStereo==0)
            return 2 + lOffset;
          else if (isStereo == 1)
            return 3 + lOffset;
        ...
```

Reply:
Addressed in Round 1 follow-up PR. trackreco/cmssw#75 https://github.com/trackreco/cmssw/pull/75/commits/ad58a25b07bb91d63a056699c00fae9c3aac1c0a